projects/components/src/user-specified-time-range-selector/user-specified-time-range-selector.component.ts

```diff
@@ -0,0 +1,38 @@
+import { ChangeDetectionStrategy, Component } from '@angular/core';
+import { ActivatedRoute, UrlSegment } from '@angular/router';
+import { LoggerService, NavigationService, TimeRange, UserSpecifiedTimeRangeService } from '@hypertrace/common';
+import { isNil } from 'lodash-es';
+
+@Component({
+  selector: 'ht-user-specified-time-range-selector',
+  template: ` <ht-time-range (timeRangeSelected)="this.onTimeRangeSelected($event)"></ht-time-range> `,
+  changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class UserSpecifiedTimeRangeSelectorComponent {
+  public constructor(
+    private readonly userSpecifiedTimeRangeService: UserSpecifiedTimeRangeService,
+    private readonly navigationService: NavigationService,
+    private readonly loggerService: LoggerService
+  ) {}
+
+  public onTimeRangeSelected(selectedTimeRange: TimeRange): void {
+    const activatedRoute = this.navigationService.getCurrentActivatedRoute();
+    const urlSegments: UrlSegment[] = activatedRoute.pathFromRoot.flatMap(activeRoute => activeRoute.snapshot.url);
+
+    if (urlSegments.length === 1 && this.shouldSavePageTimeRange(activatedRoute)) {
```

Comment:
So we only save for top level pages? Why's that? If I'm looking at the endpoint overview and change the time range, why is that any different than if I'm looking at the endpoint list and change the time range?

Reply:
Yep, that makes sense. I was thinking of loading child pages via direct nav rather than from a parent (for example, clicking on a link for endpoint details in an email), but that's a bit of a corner case we can come back to.